### ISSUE TITLE
downloaded whole season pack just to get one episode

### ISSUE THREAD

**Issue by jtwill** [SiCKRAGETV/sickrage-issues#3004]
SR Version: 	BRANCH: (master) / COMMIT: (41dfcaeaf855297d6a478355746d6b769ca8b0da) 
Python Version:	2.7.10 (default, May 26 2015, 04:16:29) [GCC 5.1.0]
linux (archlinux)

CONFIGURATION
Show was configured for either 720p HDTV or 720p WEB-DL, with archive quality of 720p WEB-DL. Also, global Ignore words were set to exclude "webrip". (Too bad WebRip is not a separate quality choice, since WebRip is frequently much lower quality than WEB-DL, but SickRage lumps WebRip in with WEB-DL so I have to put webrip into ignore to prevent SickRage from downloading webrips)

Only provider configured was torrent provider BTN. Backlog searches were enabled. Season search fallback was enabled. Season search mode was set to season packs.

WHAT HAPPENED
My library already had all episodes for the current season (in a mix of either 720p HDTV or 720p WEB-DL quality) except for the last episode, which was set to WANTED by the daily-search. Despite the episode being available on the torrent provider, it was not downloaded during the daily-search (see separate bug report about SickRage not downloading the wanted episode during dailysearch). When the backlog search eventually ran, SickRage downloaded an entire season pack in 720p HDTV quality, just to get the final episode (note: the final episode was available in single-episode format on the torrent provider). SR also replaced some of the episodes in my library that had been 720p WEB-DL with some of the 720p HDTV episodes from the season pack.

This actually happened to me for two different shows last night (Ray, Masters), both of which had exactly the same problem as just described.

WHAT SHOULD HAVE HAPPENED
SickRage should have just downloaded the single episode that was wanted, not the whole season pack. SR should not have replaced already downloaded 720p WEB-DL episodes with 720p HDTV episodes from the season pack.

LOG
https://gist.github.com/jtwill/61277780e2beaa43e0c8

**Comment by SiCKRAGEB0T** [comment]
Thanks for the issue report! To make sure issues are handled quickly, it helps if all info is there. Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [x] Post debug logs, either inline (for smaller logs) or using gist)

Please make sure you also read [how to create an issue](https://github.com/SiCKRAGETV/sickrage-issues/blob/master/README.md) and followed all the steps. 

Make the title describe your issue. Having "SR not working" or "I am getting this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the devs don't like you, but because they are on mobile or busy other fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by miigotu** [comment]
"Season search mode was set to season packs."

You are telling SR to search for season packs FIRST with that setting.
The fallback is allowing it to search for single episodes ONLY if the season pack doesnt find results.

Change your settings.

**Comment by jtwill** [comment]
If that is true, then the bug is in the description of the setting:

"when searching for complete seasons you can choose to have it look for season packs only, or choose to have it build a complete season from just single episodes."

Note that in my situation, SR should NOT have been "searching for complete seasons". It only needed 1 episode.

From the description, it says "when searching for complete seasons....". Why am I supposed to think that applies to this case, where it should only be searching for a single episode?

**Comment by miigotu** [comment]
I agreem the setting has a confusing description. Do you have a suggestion for the description that is less confusing?

**Comment by jtwill** [comment]
Huh? I still don't know what the setting is supposed to do, so I certainly cannot give a suggestion on how to explain it.

**Comment by jtwill** [comment]
I can tell you what I thought the setting was supposed to do, based on the current description.

I thought it meant that IF sickrage was looking to grab a complete season, would it do it by grabbing a season pack, or by grabbing all individual episodes in the season.

I certainly did NOT expect it to mean that SR would just grab an entire season pack when only was episode was needed.

**Comment by jtwill** [comment]
Actually, now I have another question.

In situations where SickRage does need to grab an entire season of a show (for example, I just added a new show to SR and it needs to grab the entire first season), I would prefer that SR grab a season pack rather than grabbing all the individual episodes.

But you just told me that I need to change the "Season search mode" away from "season packs" and to "episodes". But if I make that change, then won't SR just grab the season by getting all the individual episodes? Which is not what I want.

I think that there really is a bug in SR, not just in the option description. Logically, the option SHOULD behave the way it is described (and the way I interpreted it). It does not make sense to have SR behave in the way you imply.

**Comment by miigotu** [comment]
But that IS how it works, and has for as long as Ive been with the project. Yes it could be improved, but it is not a "bug".

Fallback means it will search both, the radio buttons is what to search first.
If a whole season is wanted, it will select the full season instead if fallback is enabled, even if season is not the primary method.

**Comment by jtwill** [comment]
I'm still not sure what you mean by "IS how it works". Exactly how does it work?

Are you saying that when I set it to "season pack", whenever there is a season pack available, SR will download the entire season pack, ALWAYS, even if only one episode from the season is wanted?

And if I set it to "episode", then SR will ALWAYS download individual episodes, even if ALL episodes for an entire season are wanted?

If that is so, then it  is extremely non-intuitive behavior, and seems quite contrary to the current description. Also, I am not sure why anyone would ever want to set it to "season pack" if it is going to download an entire season just to get a single episode.

By the way, yesterday DIMENSION posted the episodes of two shows (Ray, Masters) several hours before the airtime SR thought it should be. Then, when SR daily-search finally looked for them a few hours later (when SR thought they should air), SR refused to download them. Then when my backlog search finally ran at around 5am (GMT -7), the season packs had already been posted. What a mess. I can understand why SR might not want to actually search for episodes much earlier than it thinks they are going to air, but once SR has actually changed them to Wanted from Unaired, it is just terrible that SR still refuses to download them during the daily-search.
